### PR TITLE
[Mirror] Add price improvement protocol fee

### PR DESCRIPTION
**Pull request by fhenneke**
This PR mirrors the changes from https://github.com/cowprotocol/solver-rewards/pull/353.

Besides introducing price improvement fees, mirroring the changes from solver-rewards fixed some issues in order rewards.

### CODE REVIEW

**Comment by harisang** [comment]
> fixed some issues in order rewards.

Can you elaborate on what was the issue that was fixed here?

File: src/sql/orderbook/prod_order_rewards.sql [hunk 1]
```diff
@@ -110,12 +142,25 @@ order_protocol_fee_prices AS (
         opf.auction_id,
         opf.protocol_fee,
         opf.protocol_fee_token,
-        ap.price / pow(10, 18) as protocol_fee_native_price
+        CASE
```
Comment (harisang) [inline src/sql/orderbook/prod_order_rewards.sql:145]: Are these used anywhere? Or did you add them for completeness purposes?

File: src/sql/orderbook/prod_order_rewards.sql [hunk 2]
```diff
@@ -110,12 +142,25 @@ order_protocol_fee_prices AS (
...
+            WHEN opf.sell_token != opf.protocol_fee_token THEN (opf.sell_amount - opf.observed_fee) / opf.buy_amount * opf.protocol_fee
+            ELSE opf.protocol_fee
+        END AS network_fee_correction,
+        opf.sell_token as network_fee_token,
+        ap_surplus.price / pow(10, 18) as surplus_token_price,
```
Comment (harisang) [inline src/sql/orderbook/prod_order_rewards.sql:150]: would prefer to add the `native` term in the column name, e.g., `surplus_token_native_price`

File: src/sql/orderbook/prod_order_rewards.sql (hunk 1 above)
Comment (fhenneke) [reply src/sql/orderbook/prod_order_rewards.sql:145]: They are not used anywhere. This was mostly to make it more or less a copy paste from the solver-rewards repo. Unfortunately there is some deviation anyways (e.g. the solver-rewards query does not keep `order_uid` as a column till the end.

Generally I would be in favor of making the queries as close as possible. So we should imho rather add used information to solver-rewards than remove unused info here.

File: src/sql/orderbook/prod_order_rewards.sql (hunk 2 above)
Comment (fhenneke) [reply src/sql/orderbook/prod_order_rewards.sql:150]: If you think this is useful we should add it to the solver-rewards repo and copy the change here after merging over there.

**Comment by fhenneke** [comment]
I added a PR in solver-rewards on changing details of those SQL queries: https://github.com/cowprotocol/solver-rewards/pull/354

**Review by harisang — APPROVED** [review]

**Comment by harisang** [comment]
Seems i merged it although some test was failing. Sorry, misread what was failing and thought it was for another reason. PR #93 addresses this.

**Pull request merged**